lib/core/src/sdk.rs

```diff
@@ -181,24 +184,19 @@ impl LiquidSdkBuilder {
             &fingerprint_hex,
         )?;
 
-        let sync_enabled = self
-            .config
-            .sync_service_url
-            .clone()
-            .map(|_| true)
-            .unwrap_or(false);
-
         let persister = match self.persister.clone() {
             Some(persister) => persister,
             None => {
-                let persister = Arc::new(Persister::new(
-                    &working_dir,
-                    self.config.network,
-                    sync_enabled,
-                )?);
-                persister.init()?;
-                persister.replace_asset_metadata(self.config.asset_metadata.clone())?;
-                persister
+                if cfg!(not(all(target_family = "wasm", target_os = "unknown"))) {
+                    Arc::new(Persister::new_using_fs(
+                        &self.get_working_dir()?,
+                        self.config.network,
+                        self.config.sync_enabled(),
+                        self.config.asset_metadata.clone(),
+                    )?)
+                } else {
+                    panic!("Must provide a WASM-compatible persister on WASM builds")
+                }
```

Comment:
I don't think this will compile as the `cfg!` macro does not conditionally compile blocks/statements. Maybe:

```suggestion
                #[cfg(not(all(target_family = "wasm", target_os = "unknown")))]
                Arc::new(Persister::new_using_fs(
                    &self.get_working_dir()?,
                    self.config.network,
                    self.config.sync_enabled(),
                    self.config.asset_metadata.clone(),
                )?)
                #[cfg(all(target_family = "wasm", target_os = "unknown"))]
                return Err(anyhow!("Must provide a WASM-compatible persister on WASM builds"));
```

Reply:
Thanks. Applied in [074b2ec](https://github.com/breez/breez-sdk-liquid/pull/792/commits/074b2ec2ac5070bd5368ab209a9cc721067c0283)